packages/server/src/middlewares/rateLimiter.ts

```
@@ -0,0 +1,10 @@
+export class SlidingWindowRateLimiter {
+  private requests = new Map<string, number[]>();
+  isAllowed(key: string, limit: number, windowMs: number, now: number = Date.now()): boolean {
```

ESLint

[error] Method 'isAllowed' has too many parameters (4). Maximum allowed is 3.

(max-params)